control-plane/stor-port/src/types/v0/store/volume.rs

```diff
@@ -505,6 +521,14 @@ impl SpecTransaction<VolumeOperation> for VolumeSpec {
                 VolumeOperation::Resize(size) => {
                     self.size = size;
                 }
+                VolumeOperation::SetVolumeProperty(prop) => {
+                    let prop_name = prop.prop_name();
+                    if prop_name == VolumeAttr::MaxSnapshots.name() {
```

Comment:
Why not have the derivation with the value already set?

Reply:
Currently value is updated in the commit stage. In the update stage, We are not changing the value. Similar to SetReplica flow. As it is just a single value update, done at the end of the operation.